### PR TITLE
Possible fix for issue #10960

### LINKED ISSUE #10960: FlyoutItemIsVisible should accept a "BindableObject" (i.e. it should be data bound)
### Description

In my application I adjust flyout items dynamically.  Whereas "FlyoutItem.IsVisible" accepts a BindableObject, "FlyoutItemIsVisible" does not.  This seems like an oversight to me.

On a related note, changing the "FlyoutItemIsVisible" dynamically changes its order in the list.  This seems like a bug to me, but it's possible that addressing the former will also address the latter.

### Public API Changes

Microsoft.Maui.Controls.BaseShellItem.FlyoutItemIsVisible

### Intended Use-Case

Data binding my shell to a view model.

### PR DESCRIPTION
**Pull request by nogginbox**
### Description of Change

This is a potential fix suggested by @mattleibow for FlyoutItemIsVisible should accept a "BindableObject"

### Issues Fixed

Fixes #10960

### CODE REVIEW

File: src/Controls/src/Core/Shell/Shell.cs [hunk 1]
```diff
@@ -69,7 +69,7 @@ static void OnSearchHandlerPropertyChanged(BindableObject bindable, object oldVa
 		}
 
 		public static readonly BindableProperty FlyoutItemIsVisibleProperty =
-			BindableProperty.CreateAttached(nameof(IsVisible), typeof(bool), typeof(Shell), true, propertyChanged: OnFlyoutItemIsVisibleChanged);
+			BindableProperty.CreateAttached(nameof(BaseShellItem.FlyoutItemIsVisible), typeof(bool), typeof(Shell), true, propertyChanged: OnFlyoutItemIsVisibleChanged);
```
Comment (mattleibow) [inline src/Controls/src/Core/Shell/Shell.cs:72]:
```suggestion
			BindableProperty.CreateAttached("FlyoutItemIsVisible", typeof(bool), typeof(Shell), true, propertyChanged: OnFlyoutItemIsVisibleChanged);
```

**Review by mattleibow — DISMISSED** [review]
Thanks for this. The new name is correct, but for attached properties we should just use a string.

**Review by Redth — APPROVED** [review]

**Pull request merged**